packages/dashboard/src/components/widgets/tile/tile.tsx

```diff
@@ -87,6 +91,23 @@ const WidgetTile: React.FC<WidgetTileProps> = ({ children, widget, title, remove
     });
   };
 
+  const [delayLoading, setDelayLoading] = useState(false);
+  const timeout = useRef<number | null | NodeJS.Timeout>(null);
+
+  useEffect(() => {
+    clearTimeout(timeout.current as number);
+    if (isRefreshing) {
+      timeout.current = setTimeout(() => {
+        setDelayLoading(isRefreshing);
+      }, USE_CASE_DELAY_IN_SECS);
+    } else {
+      setDelayLoading(false);
+    }
+    return () => {
+      clearTimeout(timeout.current as number);
+    };
+  }, [isRefreshing]);
+
```

Comment:
This clutters the code base a bit and is re-usable. Let's pull this out into a custom hook

Reply:
resolved